subsys/bluetooth/mesh/sol_pdu_rpl_srv.c

```diff
@@ -67,8 +67,12 @@ static int item_clear(struct bt_mesh_model *mod,
 		return -EINVAL;
 	}
 
-	for (int i = 0; i < len; i++) {
-		bt_mesh_srpl_entry_clear(primary + i);
+	if (len == 0) {
+		bt_mesh_srpl_entry_clear(primary);
+	} else {
+		for (int i = 0; i < len; i++) {
```

Comment:
I guess this will be sufficient
```suggestion
		for (int i = 0; i < len || len == 0; i++) {
```

Reply:
Yeah, I thought that something like this misses some edge case but now looks ok to me. But isn't this just same as `for (int i = 0; i <= len; i++) {`?

Reply:
> `for (int i = 0; i <= len; i++) {`?
This is different, upper boundary shouldn't be included

Reply:
Oh, right, this was this edge case :)

Reply:
Sorry, I made mistake. It should be: 

`for (int i = 0; i < len || i == 0; i++) {`